service/matching/physical_task_queue_manager_test.go

```diff
@@ -116,6 +118,54 @@ func TestSyncMatchLeasingUnavailable(t *testing.T) {
 	require.True(t, sync)
 }
 
+// addTasks is a helper which adds numberOfTasks to a taskTracker
+func trackTasksHelper(tr *taskTracker, numberOfTasks int) {
+	for i := 0; i < numberOfTasks; i++ {
+		// adding a bunch of tasks
+		tr.trackTasks()
+	}
+}
+
+func TestAddTasksRate(t *testing.T) {
+	// define a fake clock and it's time for testing
+	timeSource := clock.NewEventTimeSource()
+	currentTime := time.Now()
+	timeSource.Update(currentTime)
+
+	tr := newTaskTracker(timeSource, intervalSize, totalIntervalSize)
+
+	// mini windows will have the following format : (start time, end time)
+	// (0 - 4), (5 - 9), (10 - 14), (15 - 19), (20 - 24), (25 - 29), (30 - 34), ...
+
+	// tasks should be placed in the first mini-window
+	timeSource.Advance(1 * time.Second) // time: 1 second
+	trackTasksHelper(tr, 100)
+	require.Equal(t, float32(3.2258065), tr.rate()) // 100 tasks added in (30 + 1 (current window)) seconds = 100 / 31 = 3.22
```

Comment:
expected rate in this case is 100 because the queue was alive for only 1 seconds and 100 tasks were added during that 1s.

Reply:
I agree to your point since I had had this in my previous implementation. I shall be updating the branch to reflect this for now and @dnr can correct me later if I had understood something wrong. Thanks for taking a peek!